### PR TITLE
Modify helm chart to use pod_template_file

### PR DESCRIPTION
**Pull request by dimberman**
Since we are deprecating most k8sexecutor arguments
we should use the pod_template_file when launching airflow
using the KubernetesExecutor

(cherry picked from commit 1b71da3e3717f340cb379ffa1e9fa809cb4bf0b1)

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by dimberman** [comment]
@potiuk @turbaszek any idea how I can unit test this? This is a bit tricky since I'm generating a string inside of a config_file instead of an actual yaml object that helm unittest can test

**Comment by mik-laj** [comment]
Hi.

Thank you for your work on KubernetesExecutor. I am wondering if you would be interested to add some documentation for pod_template_file?
https://github.com/apache/airflow/issues/9958
It seems to me to be a low hanging fruit right now. Many users point out to me the lack of documentation for this option.

File: chart/templates/configmap.yaml [hunk 1]
```diff
@@ -55,3 +55,11 @@ data:
   known_hosts: |
     {{ .Values.dags.gitSync.knownHosts | nindent 4 }}
 {{- end }}
+{{- if eq .Values.executor "KubernetesExecutor" }}
+  pod_template_file.yaml: |-
+{{- if .Values.podTemplateFile }}
```
Comment (ashb) [inline chart/templates/configmap.yaml]: This value name needs clarifying -- seeing the name `file` I'd expect this to be a path to the file, but if I'm reading this correctly the helm value is the contents of the file/YAML to use as the pod template, correct?

If so `.Values.podTemplate` or `.Values.executorPodTemplate` might be a better name

File: chart/values.yaml [hunk 2]
```diff
@@ -27,6 +27,8 @@ gid: 50000
 # Used for mount paths
 airflowHome: "/opt/airflow"
 
+podTemplateFilePath: ~
```
Comment (ashb) [inline chart/values.yaml]: This one isn't used anywhere.

File: chart_pod_template_file/tests/pod-template-file_test.yaml [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+---
+templates:
+  - pod-template-file.yaml
```
Comment (ashb) [inline chart_pod_template_file/tests/pod-template-file_test.yaml]: Is it not possible to have this in the "normal" airflow `chart/` and have this as `files/pod-template-file.yaml`?

Comment (dimberman) [reply chart_pod_template_file/tests/pod-template-file_test.yaml]: @ashb I wish. The problem is that helm-unittest can ONLY test files in a "templates" folder. So if we keep it there, we can't unittest it.

Comment (dimberman) [reply chart_pod_template_file/tests/pod-template-file_test.yaml]: @ashb wait I think I figured something out

Comment (ashb) [reply chart_pod_template_file/tests/pod-template-file_test.yaml]: Nice

File: airflow/kubernetes/pod_generator.py [hunk 4]
```diff
@@ -501,15 +501,16 @@ def reconcile_containers(base_containers: List[k8s.V1Container],
         )
 
     @staticmethod
-    def construct_pod(
+    def construct_pod(  # pylint: disable=R0913
```
Comment (kaxil) [inline airflow/kubernetes/pod_generator.py]:
```suggestion
    def construct_pod(  # pylint: disable=too-many-arguments
```

Comment (kaxil) [reply airflow/kubernetes/pod_generator.py]: Definitely not a concern but just more clearer

**Review by kaxil — APPROVED** [review]

**Review by danielhoherd — APPROVED** [review]

**Comment by blecx** [comment]
Hello,

is there any documentation how this works now. Can't find it in the documentation. Need this for custom images

Thank you very much for any help

**Comment by kaxil** [comment]
@blecx  - Check https://airflow.apache.org/docs/helm-chart/stable/customizing-workers.html?highlight=pod_template_file#custom-pod-template-file

**Pull request merged**